components/spotify/spotify.app.js

```diff
@@ -12,34 +33,41 @@ module.exports = {
         Authorization: `Bearer ${this.$auth.oauth_access_token}`,
       };
     },
-    async getPlaylistItems(playlist_id, params) {
-      return await axios.get(
-        `${await this._getBaseUrl()}/playlists/${playlist_id}/tracks`,
-        {
-          headers: await this._getHeaders(),
-          params,
+    async _makeRequest(method, endpoint, params) {
+      config = {
+        method,
+        url: `${await this._getBaseUrl()}/${endpoint}`,
+        headers: await this._getHeaders(),
+        params,
+      };
+      return await this.retry(config);
+    },
+    // Retry axios request if not successful
+    async retry(config, retries = 3) {
+      let response;
+      try {
+        response = await axios(config);
+        return response.data;
+      } catch(err) {
+        if (retries <= 1) {
+          throw new Error(err);
         }
-      );
+        // if rate limit is exceeded, Retry-After will contain the # of seconds to wait before retrying
+        delay = (response && response.status == 429) ? (response.headers['Retry-After']*1000) : 500;
```

Comment:
`const delay`? Or is `delay` declared somewhere else above?

Reply:
Updated.